_posts/2020-09-17-accessing-snowplow-data-in-real-time-aws/index.md

```diff
@@ -0,0 +1,123 @@
+
+
+# Tutorial
+This is a really simple tutorial of reading from the real time stream. What we're going to do is set up a lambda function to trigger when data is written to the good enriched Kinesis events stream, transform the data into JSON with the Snowplow Python Analytics SDK and log the output to CloudWatch. Data that is written to the event stream has passed through the [validation and enrichment steps](https://docs.snowplowanalytics.com/docs/understanding-your-pipeline/architecture-overview-aws/) in the diagram below. 
+![Screenshot](img/snowplow-pipeline-diagram-v2.png)
+
+## 1. Prerequisites
+If you haven't got a snowplow pipeline set up in AWS using Kinesis streams this next part is not going to make much sense. If you're unsure [check out here to get started on open source](https://docs.snowplowanalytics.com/docs/setup-snowplow-on-aws/) or [here to contact us about our managed solution](http://snowplowanalytics.com/get-started/)
+
+## 2. Set up IAM in AWS
+For this tutorial your lambda function needs a role with permission to do the following: 
+```
+"kinesis:GetRecords",
+"kinesis:GetShardIterator",
+"kinesis:DescribeStream",
+"kinesis:ListStreams",
+"logs:CreateLogGroup",
+"logs:CreateLogStream",
+"logs:PutLogEvents"
+```
+
+## 3. Create the lambda function
+Create a new Lambda function by clicking `Create Function` and give it the following properties: 
+* Name: lambda_function_payload
+* Runtime: Python 3
+* Permissions: Use existing role, and select the role you made in the previous step
+![Screenshot](img/create_function.png)
+
+## 4. Create the Python script
+Copy this Python script to a file locally. In my case, it's called lambda_function_payload.py and is inside a subdirectory called lambda_function_payload. This subdirectory is important for the next step. 
+
+```python
+import snowplow_analytics_sdk.event_transformer
+import snowplow_analytics_sdk.snowplow_event_transformation_exception
+import json
+import base64
+
+# Decode batch of lambda records from Base64 to TSV
+def decode_records(update):
+    data = []
+    if "Records" in update:
+        for record in update["Records"]:
+            if "kinesis" in record and "data" in record['kinesis']:
+                # decode base64 data in each record and append to the list
+                decoded_data = base64.b64decode(record['kinesis']['data']).decode('utf-8')
+                data.append(decoded_data)
+    return data
+
+# Entry point for the lambda function
+def lambda_handler(event, context):
+    # Uncomment this line to show the raw batch of events
```

Comment:
Update this comment to mention where you can see the output? I assume in some cloudwatch logs?